classes/Customer.php

```diff
@@ -446,6 +447,16 @@ public function getByEmail($email, $plaintextPassword = null, $ignoreGuest = tru
         return $this;
     }
 
+    /**
+     * Convert the host part of the email from punycode to utf8 (e.g,. email@xn--e1aybc.xn--p1ai -> email@тест.рф)
+     *
+     * @return string
+     */
+    public function getEmailForDisplay(): string
+    {
+        return (new InternationalizedDomainNameConverter())->emailToUtf8($this->email);
```

Comment:
I'm not sure it's a good idea to instanciate the class at each call of this method..